content/howto/extensibility/how-to-create-a-native-pluggable-widget.md

```diff
@@ -0,0 +1,1344 @@
+  - Change glue component to passthrough header caption
+  - Add containment
+  - Add toggable functionality
+    - Touchable component (dynamic component)
+  - Add icon property for expand and collapse
+  - Add collapsible property
+  - Add default styling from modeler
+  - Create design property
+
+To {do this task}, follow these steps:
+
+1. {Step 1}
+2. {Step 2}
+
+![](attachments/{sub-folder with same name as doc file}/{image filename}.png)
+
+### 3.1 Scaffolding pluggable widget project
+
+The Pluggable Widget Generator is the quickest way to start developing a pluggable widget. When you run this generator with Yeoman, Yeoman will scaffold your project folder with a folder structure and certain files recommended by Mendix.
+
+To scaffold your project folder for the group box widget, open up a terminal, navigate to the folder where you want to store your project and start the generator by executing the following command:
+
+```shell
+yo @mendix/widget GroupBox
+```
+
+The generator will ask you for some input during setup. Please provide the following information:
+
+1. Widget name: _{Your widget name}_
+2. Widget description: _{Your widget description}_
+3. Organization name: _{Your organization name}_
+4. Copyright: _{Your copyright date}_
+5. License: _{Your license}_
+6. Initial version: _{Your initial version number}_
+7. Author: _{Your author name}_
+8. Mendix Project path: _./test/MxTestProject/_
+9. Programming language: **TypeScript**
+10. Widget type: **For native mobile apps**
+11. Widget template: **Empty widget (recommended for more experienced developers)**
+12. Unit tests: **No**
+
+In the image below you can find an example of what information to provide as input.
+![Pluggable Widget Generator input example](attachments/how-to-create-a-native-pluggable-widget/pluggable-widget-generator-input-example.png)
+
+### 3.2 Creating a Mendix test project
+
+In order to test our group box widget we want to create a Mendix application that uses this widget in various ways.
+
+1. Open Mendix Studio Pro v8.2.2 to create a new test project by selecting **File > New Project** from the top menu bar. Select the starter app **Blank**, which is the last option on the last page. Then click the **Use this starting point** button, and fill in the following details in the **App Settings** dialog box:
+
+   - App name: **GroupBoxTestProject**
+   - Enable online services: **No**
+
+   After that, click the **Create app** button.
+
+   Optionally you may remove all unused custom widgets to optimize the debugging process. Select **Project > Show Project Directory in Explorer** from the Mendix Studio Pro menu and open the **widgets** folder. Then, delete all the files in this folder.
+
+   After that, close the project by selecting **File > Close Project** from the top menu bar.
+
+2. Move the contents of the Mendix test project folder into **test/MxTestProject** which is located inside your group box widget project folder.
```

Comment:
not needed if in the the create dialog is selected the correct folder... saves the move step and re-open step